### PR TITLE
Add detection for webgl support.

### PR DESCRIPTION
**Pull request by Schinkentanz**

### CODE REVIEW

File: fingerprint2.js [hunk 1]
```diff
@@ -822,6 +822,16 @@
       var elem = document.createElement("canvas");
       return !!(elem.getContext && elem.getContext("2d"));
     },
+    isWebGlSupported: function() {
+      if (!this.isCanvasSupported() || !!window.WebGLRenderingContext) {
```
Comment (Valve) [inline fingerprint2.js]: Is `!!window.WebGLRenderingContext` correct? It's double negation, returns true when supported.
I think this needs to test for positive, rather than negative, like if(this.isCanvas && !!window.WebGLRenderingContext) { true} else { false};

**Comment by Schinkentanz** [comment]
Sry. That was a typo.. But i already tried it with Modernizr:
```javascript
new Fingerprint2({
  excludeWebGL: !Modernizr.webgl
}).get(function(fingerprintHash) {
  // do stuff
});
```
The problem is, that (e.g.) the samsung galaxy s2 seems to support webgl (```Modernizr.webgl returns true```) ... but when the webgl fingerprint is generated, an error occurs: ```undefined function getShaderPrecisionFormat```. Thats why added this special webgl check.

**Comment by Valve** [comment]
Then it's a bug in the WebGL FP itself. Please use the Modernizr code to detect the availability of the WebGL and file a separate issue with getShaderPrecisionFormat. I have an old android device will test there soon.

**Comment by Valve** [comment]
@Schinkentanz , thanks. Did you try running gulp?

File: fingerprint2.js [hunk 2]
```diff
@@ -217,9 +217,19 @@
       return keys;
     },
     webglKey: function(keys) {
-      if(!this.options.excludeWebGL && this.isCanvasSupported()) {
+      var isWebGlExcluded = !!this.options.excludeWebGL,
+          isWebGlSupported = this.isWebGlSupported();
+
+      if (!isWebGlExcluded && isWebGlSupported) {
```
Comment (Valve) [inline fingerprint2.js]: I think it's a mistake. You test for support of WebGL and then log that it's not supported.
Could you also please not introduce the isWebGlExcluded variable and operate solely on original options.excludeWebGL? I think it can be done w/out adding the variable.

File: fingerprint2.js [hunk 3]
```diff
@@ -822,6 +832,24 @@
...
+      if (!this.isCanvasSupported() || !window.WebGLRenderingContext) {
+        return false;
+      }
+
+      var canvas = document.createElement("canvas"),
+          glContext = canvas.getContext && (canvas.getContext("webgl") || canvas.getContext("experimental-webgl"));
+
+      if (glContext && !glContext.getShaderPrecisionFormat) {
+        if (typeof NODEBUG === "undefined") {
+          this.log("This browser supports webgl, but lacks some functionality (getShaderPrecisionFormat not avaialable)");
```
Comment (Valve) [inline fingerprint2.js]: I think we should still do WebGL FP, even if getShaderPrecisionFormat is not available. We should just skip this part in the WebGL code for those devices. Please add a separate issue for this, and remove this check.
I'll implement the fix for Galaxy S2 right now.

**Comment by Valve** [comment]
Hey, @Schinkentanz , thanks for the code, if you fixed the code per my comments, I'd be happy to merge.

**Comment by Schinkentanz** [comment]
Should be fine now :)

**Comment by Valve** [comment]
Thanks!

**Comment by Schinkentanz** [comment]
Np :) Could you please push a new version to npm? Thanks

**Comment by Valve** [comment]
Done, I've tweaked your code a bit to add more logging, but the idea is the same. Thanks.

**Comment by Schinkentanz** [comment]
:+1:
